*** /etc/grub.d/10_linux	2019-03-18 16:11:57.000000000 +0000
--- ./10_linux	2019-07-18 02:28:53.483399553 +0000
***************
*** 110,115 ****
--- 110,123 ----
    version="$2"
    type="$3"
    args="$4"
+   suggested_title="$5"
+ 
+   title="${os}"
+   if [ "x$suggested_title" = x ] ; then
+       title="$os"
+   else
+       title="$suggested_title"
+   fi
  
    if [ -z "$boot_device_id" ]; then
        boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
***************
*** 131,137 ****
        fi
        echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-$version-$type-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
    else
!       echo "menuentry '$(echo "$os" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-simple-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
    fi      
    if [ "$quick_boot" = 1 ]; then
        echo "	recordfail" | sed "s/^/$submenu_indentation/"
--- 139,145 ----
        fi
        echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-$version-$type-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
    else
!       echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-simple-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
    fi      
    if [ "$quick_boot" = 1 ]; then
        echo "	recordfail" | sed "s/^/$submenu_indentation/"
***************
*** 339,345 ****
  
    if [ "x$is_top_level" = xtrue ] && [ "x${GRUB_DISABLE_SUBMENU}" != xy ]; then
      linux_entry "${OS}" "${version}" simple \
!     "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
  
      submenu_indentation="$grub_tab"
      
--- 347,353 ----
  
    if [ "x$is_top_level" = xtrue ] && [ "x${GRUB_DISABLE_SUBMENU}" != xy ]; then
      linux_entry "${OS}" "${version}" simple \
!     "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}" "${GRUB_PRIMARY_MENU_TITLE}"
  
      submenu_indentation="$grub_tab"
      
